CHANGELOG.md

```diff
@@ -4,10 +4,44 @@ will be documented here.
 
 ## Unreleased
 
+## v0.2.0 TBD
+
 ### Added
 - Introduce new sub-command `solang idl` which can be used for generating
   a Solidity interface file from an Anchor IDL file. This can be used for calling
   Anchor Contracts on Solana. [seanyoung](https://github.com/seanyoung)
+- Substrate: Provide specific Substrate builtins via a "substrate" file. The
+  `Hash` type from `ink!` is the first type made available for solidity contracts.
+  [xermicus](https://github.com/xermicus)
+- Substrate: Introduce the `--log-api-return-codes` CLI flag, which changes the
+  emitted code to print return codes for `seal` API calls into the debug buffer.
+  [xermicus](https://github.com/xermicus)
+
+### Changed
+- Substrate: Supported node version is now pallet contracts `v0.22.1`.
+  [xermicus](https://github.com/xermicus)
+- Introduce function name mangling for overloaded functions and constructors, so
+  that they can be represented properly in the metadata.
+  [xermicus](https://github.com/xermicus)
+- Substrate: Remove the depricated `random` builtin.
+  [xermicus](https://github.com/xermicus)
+
+### Fixed
+- Substrate: Solang is now compatible with `ink!` version 4 (beta).
+  [xermicus](https://github.com/xermicus)
+- Substrate: Switched ABI generation to use official `ink!` crates, which fixes all
+  remaining metadata regressions.
+  [extraymond](https://github.com/extraymond) and [xermicus](https://github.com/xermicus)
+- Substrate: Allow constructors to have a name, so that multiple constructors are 
+  supported, like in `ink!`.
+  [xermicus](https://github.com/xermicus)
+- Provided examples and most of the solidity code snippets in our documentation
+  are now checked for succesful compilation on the solang CI.
+  [xermicus](https://github.com/xermicus)
+- Substrate: Fix events with topics. The topic hashes generated by solang
+  contracts are now exactly the same as those generated by `ink!`.
+  [xermicus](https://github.com/xermicus)
+
```

Comment:
@xermicus can you please add this:
```
### Fixed

 - The parser does not give up after the first error, when possible.
   [salaheldinsoliman](https://github.com/salaheldinsoliman)
 - Constant expressions are checked for overflow.
   [salaheldinsoliman](https://github.com/salaheldinsoliman)
 - AddMod and MulMod were broken, this is now fixed.
   [LucasSte](https://github.com/LucasSte)

### Changed
 - The Solana target now uses Borsh encoding rather than eth abi
   encoding. This is towards making Solang contracts Anchor compatible.
   [LucasSte](https://github.com/LucasSte)

### Removed
 - The try-catch construct is no longer permitted on Solana, as it
   never worked. Any CPI error will abort the transaction.
   [seanyoung](https://github.com/seanyoung)
```